go1.23 build fail

### Describe your question

error message
```
[ERROR] run: Command 'GOOS=linux GOARCH=amd64 go build -mod=mod -o /root/open_gemini/src/openGemini/build/ts-cli -ldflags="-X github.com/openGemini/openGemini/app.Version=1.2.0 -X github.com/openGemini/openGemini/app.GitBranch=main -X github.com/openGemini/openGemini/app.GitCommit=8797d1e89fa48271f6e6428de97e2ee83d57ea99" ./app/ts-cli' failed with error: b'# github.com/openGemini/openGemini/app/ts-cli\nlink: github.com/bytedance/sonic/ast: invalid reference to encoding/json.safeSet\n'
```

go version go1.23.1 linux/amd64
related to [sonic ](https://github.com/bytedance/sonic/issues/660). Upgrade to sonic 1.12.2 may fix the problem. 

Besides, should we check the `content` type in [build.py](https://github.com/openGemini/openGemini/blob/main/build.py#L407) since `content` may be bytes.

--- a/build.py
+++ b/build.py
@@ -405,6 +405,9 @@ def build(version=None,
 def write_to_gobuild(content):
     logging.info("write to file")
 
+    if isinstance(content,bytes):
+        content = content.decode("utf-8")
+        
     if get_system_platform() != "windows":
         with open(gobuild_out, 'w') as f:
             f.write("error\n")

--- a/go.mod
+++ b/go.mod
@@ -11,7 +11,7 @@ require (
 	github.com/apache/arrow/go/v13 v13.0.0-20230630125530-5a06b2ec2a8e
 	github.com/armon/go-metrics v0.4.1
 	github.com/bits-and-blooms/bloom/v3 v3.5.0
-	github.com/bytedance/sonic v1.9.1
+	github.com/bytedance/sonic v1.12.2
 	github.com/cespare/xxhash/v2 v2.2.0
 	github.com/cockroachdb/errors v1.9.1
 	github.com/deckarep/golang-set v1.8.0
@@ -93,9 +93,12 @@ require (
 	github.com/bgentry/speakeasy v0.1.0 // indirect
 	github.com/bits-and-blooms/bitset v1.8.0 // indirect
 	github.com/bmizerany/pat v0.0.0-20170815010413-6226ea591a40 // indirect
+	github.com/bytedance/sonic/loader v0.2.0 // indirect
 	github.com/c-bata/go-prompt v0.2.6 // indirect
 	github.com/cespare/xxhash v1.1.0 // indirect
 	github.com/chenzhuoyu/base64x v0.0.0-20221115062448-fe3a3abad311 // indirect
+	github.com/cloudwego/base64x v0.1.4 // indirect
+	github.com/cloudwego/iasm v0.2.0 // indirect
 	github.com/cockroachdb/logtags v0.0.0-20211118104740-dabe8e521a4f // indirect
 	github.com/cockroachdb/redact v1.1.3 // indirect
 	github.com/davecgh/go-spew v1.1.1 // indirect

